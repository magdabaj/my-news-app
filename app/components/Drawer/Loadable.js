/**
 *
 * Asynchronously loads the component for Drawer
 *
 */

import loadable from "utils/loadable";

export default loadable(() => import("./index"));
