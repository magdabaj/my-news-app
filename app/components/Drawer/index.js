/**
 *
 * Drawer
 *
 */

import React from "react";
import PropTypes from 'prop-types';
// import styled from 'styled-components';
import './styles.css';

const Drawer = ({ tags, isDrawerOpen, selectTag }) => {
  const drawerClass = !isDrawerOpen ? 'drawer' : 'drawer drawerOpen';

  return (
      tags !== undefined ?
          <div className={drawerClass}>
            {tags.map(tag => (
                <div
                    className={'item'}
                    key={tag.id}
                   // key={tag.tag_id}
                    onClick={()=>selectTag(tag)}
                >
                  {tag.name}
                </div>
            ))}
          </div> : null

      )

};

Drawer.propTypes = {
  tags: PropTypes.array,
  isDrawerOpen: PropTypes.bool,
  selectTag: PropTypes.func,
};

export default Drawer;
