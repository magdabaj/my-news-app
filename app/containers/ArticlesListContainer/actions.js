/*
 *
 * ArticlesListContainer actions
 *
 */

import {
  GET_ALL_ARTICLES,
  GET_ARTICLES_FOR_TAG,
  GET_ARTICLES_FOR_TAG_FAILED,
  GET_ARTICLES_FOR_TAG_SUCCESS, START_ADD
} from "./constants";

export const getAllArticles = () => ({
  type: GET_ALL_ARTICLES
});

export const getArticlesForTag = tagId => ({
  type: GET_ARTICLES_FOR_TAG,
  tagId,
});

export const getArticlesForTagSuccess = articles => ({
  type: GET_ARTICLES_FOR_TAG_SUCCESS,
  articles,
});

export const getArticlesForTagFailed = error => ({
  type: GET_ARTICLES_FOR_TAG_FAILED,
  error,
});

export const startAdd = tagName => ({
  type: START_ADD,
  tagName,
});