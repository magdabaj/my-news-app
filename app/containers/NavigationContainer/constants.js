/*
 *
 * NavigationContainer constants
 *
 */

export const GET_TAGS = "app/NavigationContainer/GET_TAGS";
export const GET_TAGS_SUCCESS = "app/NavigationContainer/GET_TAGS_SUCCESS";
export const GET_TAGS_FAILED = "app/NavigationContainer/GET_TAGS_FAILED";

export const GET_ARTICLES_BY_TAG = "app/NavigationContainer/GET_ARTICLES_BY_TAG";
