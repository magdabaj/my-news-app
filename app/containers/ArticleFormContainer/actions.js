/*
 *
 * ArticleFormContainer actions
 *
 */

import {ADD_ARTICLE, ADD_ARTICLE_CANCELED, ADD_ARTICLE_ERROR, ADD_ARTICLE_SUCCESS} from "./constants";

export const addArticle = (article, tagId, user) => ({
  type: ADD_ARTICLE,
  article,
  tagId,
  user
});

export const addArticleSuccess = article => ({
  type: ADD_ARTICLE_SUCCESS,
  article,
});

export const addArticleError = error => ({
  type: ADD_ARTICLE_ERROR,
  error,
});

export const addArticleCanceled = () => ({
  type: ADD_ARTICLE_CANCELED,
});