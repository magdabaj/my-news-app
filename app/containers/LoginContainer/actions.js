/*
 *
 * LoginContainer actions
 *
 */

import {CANCEL_LOGIN, LOGIN, LOGIN_FAILED, LOGIN_SUCCESS} from "./constants";

export const login = (email, password) => ({
  type: LOGIN,
  email,
  password,
});

export const loginSuccess = user => ({
  type: LOGIN_SUCCESS,
  user,
});

export const loginFailed = error => ({
  type: LOGIN_FAILED,
  error,
});

export const cancelLogin = () => ({
  type: CANCEL_LOGIN,
});