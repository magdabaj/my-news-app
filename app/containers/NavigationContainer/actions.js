/*
 *
 * NavigationContainer actions
 *
 */

import { GET_TAGS, GET_ARTICLES_BY_TAG, GET_TAGS_FAILED, GET_TAGS_SUCCESS } from "./constants";

export const getTags = () => ({
  type: GET_TAGS
});

export const getTagsSuccess = tags => ({
  type: GET_TAGS_SUCCESS,
  tags
});

export const getTagsFailed = error => ({
  type: GET_TAGS_FAILED,
  error
});

export const getArticlesByTag = tagId => ({
  type: GET_ARTICLES_BY_TAG,
  tagId,
});
