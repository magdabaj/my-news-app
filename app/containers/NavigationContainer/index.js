/**
 *
 * NavigationContainer
 *
 */

import React, { memo, useEffect } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { compose } from "redux";

import { useInjectSaga } from "utils/injectSaga";
import { useInjectReducer } from "utils/injectReducer";
import makeSelectNavigationContainer, {
  makeSelectIsDrawerOpen,
  makeSelectSelectedTag,
  makeSelectTags
} from "./selectors";
import reducer from "./reducer";
import saga from "./saga";
import { getTags, selectTag, toggleDrawer} from './actions';
import Navigation from "../../components/Navigation";

export function NavigationContainer({...props }) {
  useInjectReducer({ key: "navigationContainer", reducer });
  useInjectSaga({ key: "navigationContainer", saga });

  useEffect( () => {
    if (props.tags.length === 0) {
      props.getTags();
    }
  }, []);

  return <Navigation {...props} />;
}

// NavigationContainer.propTypes = {
//   tags: PropTypes.array.isRequired,
//   getTags: PropTypes.func.isRequired,
//   getArticlesByTag: PropTypes.func.isRequired,
//   selectTag: PropTypes.func.isRequired,
//   toggleDrawer: PropTypes.func.isRequired,
//   selectedTag: PropTypes.object,
//   isDrawerOpen: PropTypes.bool.isRequired,
// };

const mapStateToProps = createStructuredSelector({
  navigationContainer: makeSelectNavigationContainer(),
  tags: makeSelectTags(),
  selectedTag: makeSelectSelectedTag(),
  isDrawerOpen: makeSelectIsDrawerOpen(),
});

function mapDispatchToProps(dispatch) {
  return {
    getTags: () => dispatch(getTags()),
    selectTag: selectedTag => dispatch(selectTag(selectedTag)),
    toggleDrawer: () => dispatch(toggleDrawer()),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps
);

export default compose(
  withConnect,
  memo
)(NavigationContainer);
