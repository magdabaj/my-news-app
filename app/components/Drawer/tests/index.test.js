/**
 *
 * Tests for Drawer
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from "react";
import { render, fireEvent } from "react-testing-library";
import 'jest-dom/extend-expect'; // add some helpful assertions
import 'react-testing-library/cleanup-after-each'

import Drawer from "../index";
import {selectTag} from "../../../containers/NavigationContainer/actions";

const tags = [
  {
    name: 'saga',
    id: 1
  },
  {
    name: 'redux',
    id: 2
  }
];

jest.mock("../../../containers/NavigationContainer/actions")

describe("<Drawer />", () => {
  it("Expect to not log errors in console", () => {
    const spy = jest.spyOn(global.console, "error");
    render(<Drawer />);
    expect(spy).not.toHaveBeenCalled();
  });

  it("Expect to show all tags", () => {
    const {getByRole} = render(<Drawer tags={tags} isDrawerOpen={true} selectTag={selectTag}/>);
    expect(getByRole(/drawerClass/i)).toBeInTheDocument()
  });

  it("Expect to fire selectedTag action", () => {
    const {getByText} = render(<Drawer tags={tags} isDrawerOpen={true} selectTag={selectTag}/>);
    expect(getByText(/redux/i)).toBeInTheDocument();
    expect(getByText(/saga/i)).toBeInTheDocument();
    fireEvent.click(getByText(/redux/i));
    expect(selectTag).toBeCalledWith({
      id: 2,
      name: 'redux'
    });
  })

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it("Should render and match the snapshot", () => {
    const {
      container: { firstChild }
    } = render(<Drawer />);
    expect(firstChild).toMatchSnapshot();
  });
});
