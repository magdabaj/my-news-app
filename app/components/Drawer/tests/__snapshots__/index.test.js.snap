// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> Should render and match the snapshot 1`] = `null`;
